sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -352,8 +353,6 @@ class Analyzer(override val catalogManager: CatalogManager) extends RuleExecutor
       // operators may need null handling as well, so we should run these two rules repeatedly.
       HandleNullInputsForUDF,
       UpdateAttributeNullability),
-    Batch("UDF", Once,
-      ResolveEncodersInUDF),
```

Comment:
hmm, shall we move the MergeInto rewrite rule after `ResolveEncodersInUDF` instead?

Reply:
IIRC we need to run `ResolveEncodersInUDF` after the `ScalaUDF Null Handling` batch

Reply:
Okay